### ISSUE TITLE
Running with --nolegacy_external_runfiles?

### ISSUE THREAD

**Issue by fsouza** [bazelbuild/bazel-gazelle#1009]
### What version of gazelle are you using?

v0.23.0

### What version of rules_go are you using?

v0.26.0

### What version of Bazel are you using?

4.0.0

### Does this issue reproduce with the latest releases of all the above?

Yes

### What operating system and processor architecture are you using?

Darwin ... 19.6.0 Darwin Kernel Version 19.6.0: Tue Jan 12 22:13:05 PST 2021; root:xnu-6153.141.16~1/RELEASE_X86_64 x86_64

### What did you do?

We have a configuration that uses the option `--nolegacy_external_runfiles` and running that, gazelle can't find the go tool.

Repro repo: https://github.com/fsouza/bazel-nolegacy_external_runfiles-repro

To reproduce:

1. clone the repo
2. run `bazel run //:gazelle`

### What did you expect to see?

`Build completed successfully` (or something like that)

### What did you see instead?

```
％ bazel run //:gazelle
INFO: Build option --legacy_external_runfiles has changed, discarding analysis cache.
INFO: Analyzed target //:gazelle (0 packages loaded, 7408 targets configured).
INFO: Found 1 target...
Target //:gazelle up-to-date:
  bazel-bin/gazelle-runner.bash
  bazel-bin/gazelle
INFO: Elapsed time: 0.628s, Critical Path: 0.19s
INFO: 14 processes: 9 internal, 5 darwin-sandbox.
INFO: Build completed successfully, 14 total actions
INFO: Build completed successfully, 14 total actions
grep: MANIFEST: No such file or directory
/private/var/tmp/_bazel_fsouza/3a1b937549007976bdc312caeaa6a5b4/execroot/__main__/bazel-out/darwin-fastbuild/bin/gazelle: warning: could not locate GOROOT used by rules_go
```

I'm not sure if this issue belongs here or in rules_go, since gazelle retrieve the tool path from rules_go, but I figured opening this issue wouldn't hurt. Please let me know if further information is needed or if this should be moved elsewhere.

**Comment by jayconrod** [comment]
This is at least an issue with Gazelle's startup script. It may also be an issue in Gazelle itself and in `github.com/bazelbuild/rules_go/go/tools/bazel`.

**Comment by fsouza** [comment]
Yeah, I wasn't sure if this was a rules_go or bazel issue, I was able to track it to this line: https://github.com/bazelbuild/bazel-gazelle/blob/d1d6af75af78da00e0dbf12d5baeb2eea0b212ec/internal/gazelle.bash.in#L25

Which in my case gets generated to something like:

```sh
GOTOOL='external/go_sdk/bin/go'
```

But if I look inside the gazelle.runfiles folder, I see the following structure:

```
％ tree go_sdk
go_sdk
└── bin
    └── go -> /private/var/tmp/_bazel_fsouza/3a1b937549007976bdc312caeaa6a5b4/external/go_sdk/bin/go

1 directory, 1 file
```

Maybe something isn't linked properly? idk

**Comment by jvolkman** [comment]
It seems [this part](https://github.com/bazelbuild/bazel-gazelle/blob/5b8616dbb7dad825c61ffbb12ea9f75622568657/internal/gazelle.bash.in#L35-L39) is wrong:
```
  runfile=$(echo "$runfile" | sed -e 's!^\(\.\./\|external/\)!!')
  if grep -q "^$runfile" MANIFEST; then
    grep "^$runfile" MANIFEST | head -n 1 | cut -d' ' -f2
    return
  fi
```

The section above never gets executed if `--nolegacy_external_runfiles` isn't set, because there is a symlink named `gazelle.runfiles/my_workspace_name/external/go_sdk/bin/go`.

But with the option specified, the `gazelle.runfiles/my_workspace_name/external` isn't created, so the section that greps MANIFEST is executed.

The issue is that the working directory when bazel-running `gazelle` is `gazelle.runfiles/my_workspace_name/`, but `MANIFEST` exists in the parent directory at `gazelle.runfiles/MANIFEST`.

The [bash runfiles init snippet](https://github.com/bazelbuild/bazel/blob/master/tools/bash/runfiles/runfiles.bash#L54-L63) uses `$0.runfies_manifest` to get the manifest data, which would evaluate to `gazelle.runfiles_manifest`. This file seems to have the same contents as `gazelle.runfiles/MANIFEST`.

**Comment by uhthomas** [comment]
@jvolkman This still seems to be happening with https://github.com/bazelbuild/bazel-gazelle/commit/757e291d1befe9174fb1fcf0d9ade733cbb6b904.

**Comment by jvolkman** [comment]
@uhthomas are you on macos? Looks like there's another problem on [this line](https://github.com/jvolkman/bazel-gazelle/blob/master/internal/gazelle.bash.in#L35) just prior to the fix in #1247.

on macos:
```shell
echo external/go_sdk/bin/go | sed -e 's!^\(\.\./\|external/\)!!'
external/go_sdk/bin/go
```

vs. linux:
```shell
echo external/go_sdk/bin/go | sed -e 's!^\(\.\./\|external/\)!!'
go_sdk/bin/go
```
